### PR TITLE
add test case for not setting schema history to empty

### PR DESCRIPTION
**Pull request by Haijuncao**
Add test case to make sure schema history can not be set to empty.

### CODE REVIEW

File: src/dbnode/storage/database.go [hunk 1]
```diff
@@ -236,10 +236,18 @@ func (d *db) updateSchemaRegistry(newNamespaces namespace.Map) error {
 		curSchema, err := schemaReg.GetLatestSchema(metadata.ID())
 		if curSchema != nil {
 			curSchemaId = curSchema.DeployId()
+			if len(curSchemaId) == 0 {
+				d.log.Warn("can not update namespace schema to with empty deploy ID", zap.Stringer("namespace", metadata.ID()),
```
Comment (richardartoul) [inline src/dbnode/storage/database.go]: "can not update namespace schema with an empty deploy ID"

File: src/dbnode/storage/database.go [hunk 2]
```diff
@@ -236,10 +236,18 @@ func (d *db) updateSchemaRegistry(newNamespaces namespace.Map) error {
...
+					zap.String("current schema id", curSchemaId))
```
Comment (richardartoul) [inline src/dbnode/storage/database.go]: I think we don't usually put spaces in key names do we?

File: src/dbnode/storage/database.go [hunk 3]
```diff
@@ -236,10 +236,18 @@ func (d *db) updateSchemaRegistry(newNamespaces namespace.Map) error {
...
+				merr.Add(fmt.Errorf("can not updating namespace(%v) schema with empty deploy ID", metadata.ID().String()))
```
Comment (richardartoul) [inline src/dbnode/storage/database.go]: "can not update"

File: src/dbnode/storage/namespace.go [hunk 4]
```diff
@@ -373,6 +373,7 @@ func (n *dbNamespace) SetSchemaHistory(value namespace.SchemaHistory) {
 	schema, ok := value.GetLatest()
 	if !ok {
 		n.log.Error("can not update namespace schema to empty", zap.Stringer("namespace", n.ID()))
+		return
```
Comment (richardartoul) [inline src/dbnode/storage/namespace.go]: Should we add a test for this too? or is it covered by the other test you just added?

File: src/dbnode/storage/database.go [hunk 5]
```diff
@@ -236,10 +236,18 @@ func (d *db) updateSchemaRegistry(newNamespaces namespace.Map) error {
...
+			}
 		}
 		// Log schema update.
 		latestSchema, found := metadata.Options().SchemaHistory().GetLatest()
 		if !found {
+			d.log.Warn("can not update namespace schema to empty", zap.Stringer("namespace", metadata.ID()),
+				zap.String("current schema", curSchemaId))
 			merr.Add(fmt.Errorf("can not updating namespace(%v) schema to empty", metadata.ID().String()))
 			continue
 		} else {
```
Comment (richardartoul) [inline src/dbnode/storage/database.go]: Not really related to your change but does this `else` do anything? Since the previous branch continues wonder if we can just drop it and save a level of indentation

File: src/dbnode/storage/database.go (hunk 2 above)
Comment (Haijuncao) [reply src/dbnode/storage/database.go]: why not? I have been doing that. It seem to work fine.

File: src/dbnode/storage/database.go (hunk 1 above)
Comment (Haijuncao) [reply src/dbnode/storage/database.go]: ok

File: src/dbnode/storage/database.go (hunk 5 above)
Comment (Haijuncao) [reply src/dbnode/storage/database.go]: good catch... maybe continue was added later, which render else useless.

File: src/dbnode/storage/database.go (hunk 2 above)
Comment (richardartoul) [reply src/dbnode/storage/database.go]: it works, I think we just usually use camelCase everywhere else. I'd rather it stay consistent

File: src/dbnode/storage/namespace.go (hunk 4 above)
Comment (Haijuncao) [reply src/dbnode/storage/namespace.go]: yes, it is covered by the test

**Comment by Haijuncao** [comment]
closed as #1722
